iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/XMLImporter.java

```diff
@@ -58,32 +59,43 @@ public String getComment() {
     }
 
     protected boolean exists(ReadOnlyDataSource dataSource, String ext) throws IOException {
-        try {
-            if (ext != null) {
-                try (InputStream is = dataSource.newInputStream(null, ext)) {
-                    // check the first root element is network and namespace is IIDM
-                    XMLStreamReader xmlsr = getXMLInputFactory().createXMLStreamReader(is);
-                    try {
-                        while (xmlsr.hasNext()) {
-                            int eventType = xmlsr.next();
-                            if (eventType == XMLStreamConstants.START_ELEMENT) {
-                                String name = xmlsr.getLocalName();
-                                String ns = xmlsr.getNamespaceURI();
-                                return NetworkSerDe.NETWORK_ROOT_ELEMENT_NAME.equals(name)
-                                        && (Stream.of(IidmVersion.values()).anyMatch(v -> v.getNamespaceURI().equals(ns))
-                                        || Stream.of(IidmVersion.values()).filter(v -> v.compareTo(IidmVersion.V_1_7) >= 0).anyMatch(v -> v.getNamespaceURI(false).equals(ns)));
-                            }
-                        }
-                    } finally {
-                        cleanClose(xmlsr);
+        if (ext == null) {
+            return false;
+        }
+
+        try (InputStream is = dataSource.newInputStream(null, ext)) {
+            XMLStreamReader xmlsr = getXMLInputFactory().createXMLStreamReader(is);
+            try {
+                return isValidNetworkRoot(xmlsr);
+            } finally {
+                cleanClose(xmlsr);
+            }
+        } catch (XMLStreamException e) {
+            return false; // not a valid XML file
+        }
+    }
+
+    private boolean isValidNetworkRoot(XMLStreamReader xmlsr) throws XMLStreamException {
+        while (xmlsr.hasNext()) {
+            if (xmlsr.next() == XMLStreamConstants.START_ELEMENT) {
+                String name = xmlsr.getLocalName();
+                String ns = xmlsr.getNamespaceURI();
+
+                if (!ns.isEmpty()) {
+                    String xmlIidmVersion = ns.substring(ns.lastIndexOf('/') + 1);
+                    if (IidmVersion.compareVersions(xmlIidmVersion, CURRENT_IIDM_VERSION.toString()) > 0) {
```

Comment:
Instead of comparing the strings of the 2 versions, it would be safer to try to load the version corresponding to the namespace, and if it fails to throw the exception with your message (in which you can also add the version that was encountered).

Reply:
You can load the IidmVersion corresponding to a string with `IidmVersion.of(String version, String separator)`.
It throws an exception containing `version` if no `IidmVersion` value is found (so the version is not supported). You can just catch this exception and create your exception (with your message untouched) using the caught exception as cause.